slo/promql.go

```diff
@@ -139,6 +146,44 @@ func (o Objective) QueryErrors(window model.Duration) string {
 			grouping:      o.Indicator.Latency.Grouping,
 		}.replace(expr)
 
+		return expr.String()
+	case LatencyNative:
+		expr, err := parser.ParseExpr(`sum by (grouping) (metric{matchers="total"}) - sum by (grouping) (errorMetric{matchers="errors"})`)
+		if err != nil {
+			return ""
+		}
+
+		metric := increaseName(o.Indicator.LatencyNative.Total.Name, window)
+		matchers := cloneMatchers(o.Indicator.LatencyNative.Total.LabelMatchers)
+		for i, m := range matchers {
+			if m.Name == labels.MetricName {
+				matchers[i].Value = metric
+				break
+			}
+		}
+		matchers = append(matchers, &labels.Matcher{Type: labels.MatchEqual, Name: "slo", Value: o.Name()})
+
+		// Add the matcher {le=""} to select the recording rule that summed up all requests
+		totalMatchers := append(cloneMatchers(matchers), &labels.Matcher{
+			Type:  labels.MatchEqual,
+			Name:  labels.BucketLabel,
+			Value: "",
+		})
```

Comment:
Looks strange to have an empty `le=""` matcher here 🤔 

Do Native Histograms have the `le` label? Could we just not add it at all?

Reply:
Native histograms do not have an `le` label. I guess the label comes from the recording rule.

Reply:
Maybe use `le="+Inf"` for that recording rule?

Reply:
Yes, these labels are added through recording rules, just like we have them with the normal histograms so far. 

The `le=""` matcher actually selects the time series where no `le` label is present. I can see how that is confusing though. I like the idea of using the `le="+Inf"` a lot! :100: 

Given this is a change for both `latency` and `latencyNative` indicators I'll make that change outside of this PR.